semantics/src/traversal/functions.rs

```diff
@@ -106,14 +106,28 @@ fn func_stmt(
         fe::FuncStmt::While { .. } => unimplemented!(),
         fe::FuncStmt::If { .. } => unimplemented!(),
         fe::FuncStmt::Assert { .. } => unimplemented!(),
-        fe::FuncStmt::Expr { .. } => unimplemented!(),
+        fe::FuncStmt::Expr { .. } => expr(scope, context, stmt),
         fe::FuncStmt::Pass => unimplemented!(),
         fe::FuncStmt::Break => unimplemented!(),
         fe::FuncStmt::Continue => unimplemented!(),
         fe::FuncStmt::Revert => Ok(()),
     }
 }
 
+fn expr(
+    scope: Shared<FunctionScope>,
+    context: Shared<Context>,
+    stmt: &Spanned<fe::FuncStmt>,
+) -> Result<(), SemanticError> {
+    if let fe::FuncStmt::Expr { value } = &stmt.node {
+        let spanned = spanned_expression(&stmt.span, value);
+        // TODO: Whats the reason the expr isn't spanned?
```

Comment:
If a child node fits entirely within the parent, we don't give them a new span. I guess this would prevent us from having duplicate spans, but I'm not sure if this is necessary.

Reply:
I don't fully understand what you mean by that

> fits entirely within the parent

What is the difference in terms of spans for these two snippets:

```
def foo():
    return 1
```
or

```
def foo():
    self.bar()
```

What makes `self.bar()` *fit entirely within the parent* but `return 1` doesn't?

Reply:
In the case of a return statement, the expression that is nested within it (`1`) does not occupy the entire span of `return 1`, so it gets its own span. The span for `return 1` would be something like `(17, 24)`, while the span of `1` would be `(23, 24)`

In the case of the expression statement, the expression occupies the entire span of the parent node (`self.bar()` is `self.bar()`), which will always be true for expression statements,  so we don't bother assigning the same span to the expression.

Reply:
Oh! Now I understand. Thanks for taking the time to explain it in that detailed way :+1: